ffi/diplomat/src/timezone.rs

```diff
@@ -0,0 +1,98 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use icu_timezone::CustomTimeZone;
+use icu_timezone::GmtOffset;
+use icu_timezone::MetaZoneId;
+use icu_timezone::TimeZoneBcp47Id;
+use icu_timezone::ZoneVariant;
+
+#[diplomat::bridge]
+pub mod ffi {
+    use crate::errors::ffi::ICU4XError;
+    use crate::tinystr::ffi::TinyAsciiStr2;
+    use crate::tinystr::ffi::TinyAsciiStr4;
+    use crate::tinystr::ffi::TinyAsciiStr8;
+    use alloc::boxed::Box;
+    use core::str::FromStr;
+    use diplomat_runtime::DiplomatResult;
+    use icu_timezone::CustomTimeZone;
+
+    pub struct ICU4XGmtOffset {
+        pub(super) data: i32,
+    }
+
+    pub struct ICU4XTimeZoneBcp47Id {
+        pub id: TinyAsciiStr8,
```

Comment:
thought: hmm, note that this will lead to people doing things like `{ gmt_offset: { id: TinyAsciiStr8.new(...)}, .... }` etc

Seems a bit heavy to have this much nesting? I don't think we get much from the typed wrappers over FFI and they're not free in JS.